### PR TITLE
83 Cleanup codebase and use execve for subprocess launching

### PR DESCRIPTION
**Pull request by sgnn7**
- Codebase cleaned up to use go modules
- Acceptance tests fixed up to be self-contained
- Made subprocess launching use execve (#83)

### CODE REVIEW

**Comment by sgnn7** [comment]
CC @jepperson2

File: Jenkinsfile [hunk 1]
```diff
@@ -17,31 +17,31 @@ pipeline {
...
       steps {
-        sh 'cp ./dist/linux_amd64/summon summon'
-        dir('acceptance') {
-          sh 'make'
-        }
-        // TODO: remove need to sudo here
-        sh 'sudo chown -R jenkins:jenkins .'
-        // TODO: collect the acceptance test results
+        sh './test_acceptance'
+        junit 'output/acceptance/*.xml'
```
Comment (dustinmm80) [inline Jenkinsfile]: artifact collection steps need to be wrapped in an `always` block, or if the previous `sh` step fails no results will be collected.

**Comment by sgnn7** [comment]
@dsbyrne Moved the `command` into `internal/command`. Sadly I don't have enough time to fix up the exec script directories before leaving for the conf so I think that might wait until a bit later so this is CR ready again.

**Comment by dsbyrne** [comment]
💯

**Pull request merged**